sycl/source/detail/pi_opencl.cpp

```diff
@@ -241,6 +241,39 @@ pi_result OCL(piSamplerCreate)(pi_context context,
   return error_code;
 }
 
+pi_result OCL(piextGetDeviceFunctionPointer)(pi_device device,
+                                             pi_program program,
+                                             const char *func_name,
+                                             pi_uint64 *function_pointer_ret) {
+  pi_platform platform;
+  PI_CALL(piDeviceGetInfo(device, PI_DEVICE_INFO_PLATFORM, sizeof(platform),
+                          &platform, NULL));
+  using FuncT =
+      cl_int(CL_API_CALL *)(cl_device_id, cl_program, const char *, cl_ulong *);
+
+  // TODO: add check that device supports corresponding extension
+  FuncT func_ptr =
+      reinterpret_cast<FuncT>(clGetExtensionFunctionAddressForPlatform(
```

Comment:
> Do I understand it right that you are looking for the clGetExtensionFunctionAddressForPlatform each time user asks for function pointer. 

Yes, you are right. 

> If so, is there any plans to improve it?

Not sure that I want to do this right now (at least in scope of this PR) if it requires significant amount of changes. There is a TODO about mapping host function pointer directly to a device function pointer without using a function name - it potentially will bring some changes to the implementation and improvement you mentioned can be done during this phase.

However, if you or @smaslov-intel have some suggestions how it can be done in existing infrastructure - it would be helpful